src/openrct2/actions/GameActionResult.h

```diff
@@ -82,13 +87,21 @@ namespace GameActions
         // is still just uint32_t, this guarantees the data is associated with the correct type.
         template<typename T> void SetData(const T&& data)
         {
-            ResultData = std::make_shared<T>(data);
+#ifdef __ANDROID__
+            ResultDataPtr = std::make_shared<T>(data);
+#else
+            ResultData = std::forward<const T&&>(data);
+#endif
         }
 
         template<typename T> T GetData() const
         {
-            T* res = static_cast<T*>(ResultData.get());
-            return *res;
+#ifdef __ANDROID__
+            return *static_cast<T*>(ResultDataPtr.get());;
+#else
+            // This function will throw std::bad_any_cast if the type mismatches.
```

Comment:
I think this is alright to remove. It's on the `any_cast` documentation and you already explain above.
```suggestion
```

Reply:
I’d like to keep it anyway.